docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,503 @@
+    CHIP_ERROR SetSetupPasscode(uint32_t setupPasscode) override;
+
+    // ===== Members functions that implement the DeviceInstanceInfoProvider
+    CHIP_ERROR GetVendorName(char * buf, size_t bufSize) override;
+    CHIP_ERROR GetVendorId(uint16_t & vendorId) override;
+    CHIP_ERROR GetProductName(char * buf, size_t bufSize) override;
+    CHIP_ERROR GetProductId(uint16_t & productId) override;
+    CHIP_ERROR GetSerialNumber(char * buf, size_t bufSize) override;
+    CHIP_ERROR GetManufacturingDate(uint16_t & year, uint8_t & month, uint8_t & day) override;
+    CHIP_ERROR GetHardwareVersion(uint16_t & hardwareVersion) override;
+    CHIP_ERROR GetHardwareVersionString(char * buf, size_t bufSize) override;
+    CHIP_ERROR GetRotatingDeviceIdUniqueId(MutableByteSpan & uniqueIdSpan) override;
+```
+
+You can read the factory data set from the device's Flash memory in different ways depending on purpose and format. In the nRF Connect example, the factory data is stored in CBOR format. Then a device uses the [Factory Data Parser](../../src/platform/nrfconnect/FactoryDataParser.h) to read out raw data, decode them, and store them in the `FactoryData` structure. The [Factor Data Provider](../../src/platform/nrfconnect/FactoryDataProvider.c) implementation uses this parser to get all needed factory data parameters and provide them to the Matter core.
+
+The last thing to do is add a newly created parser and accessor to the build system by replacing the default ones. To do that, we recommend putting new implementations into *connectedhomeip-root/src/platform/nrfconnect* directory, editing the [BUILD.gn](../../src/platform/nrfconnect/BUILD.gn), finding out section:
+```
+  if (chip_enable_factory_data) {
+    sources += [
+      "FactoryDataParser.c",
+      "FactoryDataParser.h",
+      "FactoryDataProvider.cpp",
+      "FactoryDataProvider.h",
+    ]
+  }
+``` 
+And replacing existing `sources` with own file names.
+
+<hr>
+<a name="Generating factory data"></a>
+
+
+## Enabling factory data support
+
+By default, factory data support is disabled in all nRF Connect examples. An nRF Connect device uses pre-defined parameters from Matter Core, which the user should not change. To start using factory data stored in the Flash memory and use the nRF Connect **Factory Data Provider** build an example with following option (replace `nrf52840dk_nrf52840` with your board name):
+
+```
+$ west build -b nrf52840dk_nrf52840 -- -DCONFIG_CHIP_FACTORY_DATA=y
+```
+
+## Generating factory data
+
+This section is an example of generating factory data using two nRF Connect python scripts. The first one allows creating a JSON file containing a user-friendly representation of factory data. The second one uses a previously created JSON file to create a factory data partition and save it to a Hex file. After these operations, program a hex file containing factory data partition into the device's Flash memory.
```

Comment:
```suggestion
This section is an example of generating factory data using two nRF Connect Python scripts. The first one allows creating a JSON file containing a user-friendly representation of factory data. The second one uses a previously created JSON file to create a factory data partition and save it to a HEX file. After these operations, program a hex file containing factory data partition into the device's flash memory.
```
or am I wrong and we should capitalize "Flash" @greg-fer?

Reply:
No no, we should not.